auth/src/main/java/com/firebase/ui/auth/ui/email/RegisterEmailActivity.java

```diff
@@ -129,13 +165,129 @@ private void setUpTermsOfService() {
+                    checkAccountExists(email);
+                    mPrevEmail = email;
+                }
+            }
+        } else if (id == R.id.name) {
+            mNameValidator.validate(mNameEditText.getText());
+        } else if (id == R.id.password) {
+            mPasswordFieldValidator.validate(mPasswordEditText.getText());
+        }
+    }
+
+    public void checkAccountExists(final String email) {
+        if (!TextUtils.isEmpty(email)) {
+            mActivityHelper.getFirebaseAuth()
+                    .fetchProvidersForEmail(email)
+                    .addOnFailureListener(
+                            new TaskFailureLogger(TAG, "Error fetching providers for email"))
+                    .addOnCompleteListener(
+                            new OnCompleteListener<ProviderQueryResult>() {
+                                @Override
+                                public void onComplete(@NonNull Task<ProviderQueryResult> task) {
+                                    if (task.isSuccessful()) {
+                                        List<String> providers = task.getResult().getProviders();
+                                        if (providers != null && !providers.isEmpty()) {
+                                            // Account does exist
+                                            String provider = providers.get(0);
+                                            if (provider.equalsIgnoreCase(EmailAuthProvider.PROVIDER_ID)) {
+                                                Intent signInIntent = SignInActivity.createIntent(
+                                                        mActivityHelper.getApplicationContext(),
+                                                        mActivityHelper.getFlowParams(),
+                                                        email);
+                                                mActivityHelper.startActivityForResult(signInIntent,
+                                                                                       RC_SIGN_IN);
+                                            } else {
+                                                Intent intent = WelcomeBackIdpPrompt.createIntent(
+                                                        mActivityHelper.getApplicationContext(),
+                                                        mActivityHelper.getFlowParams(),
+                                                        provider,
+                                                        null,
+                                                        email);
+                                                mActivityHelper.startActivityForResult(intent,
+                                                                                       RC_WELCOME_BACK_IDP);
+                                            }
+                                        }
+                                    } // TODO: 11/23/2016 what happens if we fail?
```

Comment:
Thoughts? We didn't do anything before, but I think we should display a message or at least do something.